container/container.go

```diff
@@ -463,6 +465,9 @@ func AttachStreams(ctx context.Context, streamConfig *runconfig.StreamConfig, op
 			}
 			<-done
 		}
+		if detached {
```

Comment:
I'll suggest we still use additional detach channel, because `detach` is actually not an error, it's normal situation. Although we can use special error to report detach event, it's easily to be mixed up with unknown error. From this perspective, `detach` error and other errors are kinds of conflict.
What do you think? @cpuguy83